### PR TITLE
Issue #1753: UI Test exitAppByBackButton Fails on 4K

### PR DESCRIPTION
**Pull request by kglazko**
### Pull Request checklist

- [ ] This PR includes thorough **tests** or an explanation of why it does not
- [ ] This PR includes a **CHANGELOG entry** or does not need one
- [ ] I have considered adding **QA labels** on the associated issue (not this PR; `qa-ready` or `qa-denied`)

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/tv/firefox/FirefoxTestRunner.kt [hunk 1]
```diff
@@ -40,6 +45,14 @@ class FirefoxTestRunner : AndroidJUnitRunner() {
         fakeServiceLocator.swapIfNotNull()
 
         super.onCreate(arguments)
+
+        ActivityLifecycleMonitorRegistry.getInstance().addLifecycleCallback(object : ActivityLifecycleCallback {
+            override fun onActivityLifecycleChanged(activity: Activity, stage: Stage) {
+                if (stage === Stage.PRE_ON_CREATE) {
+                    activity.window.addFlags( FLAG_KEEP_SCREEN_ON)
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/tv/firefox/FirefoxTestRunner.kt]: We should add a comment explaining why this is needed. As well, what effect if any this has on simulator and other devices.

File: app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt [hunk 2]
```diff
@@ -48,7 +48,10 @@ class NavigationOverlayRobot {
     // against any of its views
     fun assertOverlayIsOpen() = urlBar().assertIsDisplayed()
 
-    fun assertCanGoBack(canGoBack: Boolean) = backButton().assertIsEnabled(canGoBack)
+    fun assertCanGoBack(canGoBack: Boolean) {
+        device.wait(Until.findObject(By.res(R.id.tileContainer.toString())), 5000)
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt]: CC @Baron-Severin 

I'm surprised that we have to add a manual wait. Is there any way to add inherent waits?

Comment (Baron-Severin) [reply app/src/androidTest/java/org/mozilla/tv/firefox/ui/robots/NavigationOverlayRobot.kt]: I'm surprised too.  I believe there are already waits baked into Espresso for things like animations and such.  The back button activating is triggered by changes to the A-C `Session` object, so my best guess would be that those changes are not fast enough to be covered by the built-in system.  But this is really all conjecture, tbh.

**Comment by liuche** [comment]
Looks like this failed on a lint error. Is this bug still a problem? If not we should close it.

**Comment by npark-mozilla** [comment]
@kglazko Hi Kate, let's work on updating this PR and merging it soon, since it's been a while

**Pull request closed without merging**